From 0975f7aae499c2668bd2329e6370e866ca7c84be Mon Sep 17 00:00:00 2001
From: "Yue, VincentX" <vincentx.yue@intel.com>
Date: Wed, 27 Dec 2017 16:29:03 +0800
Subject: [PATCH 09/16] Fix SDcard media files not display after copying from
 internal storage

Due to don't scan the new files in sdcard, lead to the
external sdcard media database don't update.

Test: Copy image/video files from internal to SD card through Setting

Change-Id: I40caa739d46489796d755185bc13b4f9511c202a
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-53792
Signed-off-by: Yue, VincentX <vincentx.yue@intel.com>
---
 core/java/com/android/internal/content/FileSystemProvider.java | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/java/com/android/internal/content/FileSystemProvider.java b/core/java/com/android/internal/content/FileSystemProvider.java
index 4b80a5f..e71257c 100644
--- a/core/java/com/android/internal/content/FileSystemProvider.java
+++ b/core/java/com/android/internal/content/FileSystemProvider.java
@@ -180,6 +180,12 @@ public abstract class FileSystemProvider extends DocumentsProvider {
                     throw new IllegalStateException("Failed to touch " + file);
                 }
                 childId = getDocIdForFile(file);
+                final File visibleFile = getFileForDocId(childId, true);
+                // If the external storage's mountFlag is not visible, the visibleFile will be null.
+                if (visibleFile != null) {
+                    Log.d(TAG, "start scanFile");
+                    scanFile(visibleFile);
+                }
             } catch (IOException e) {
                 throw new IllegalStateException("Failed to touch " + file + ": " + e);
             }
-- 
1.9.1

